### ISSUE TITLE
Random touch read

### ISSUE THREAD

**Issue by fundix** [Xinyuan-LilyGO/T-Display-S3-Long#34]
I need to solve the random clicking on the display. I have several boards and they all do it at different intervals. the click area is always just from this area
```
const uint16_t MIN_X = 530;
const uint16_t MAX_X = 600;
const uint16_t MIN_Y = 100;
const uint16_t MAX_Y = 150;
```

**Comment by fundix** [comment]
@LilyGO do you have any solution for this problem

**Comment by Llgok** [comment]
@fundix 
Hello, may I ask if the screen randomly clicks automatically when the finger is not touching it?

**Comment by fundix** [comment]
> @fundix Hello, may I ask if the screen randomly clicks automatically when the finger is not touching it?

Yes, we have several Long units, the one I have on my desk now gerates random clicks in this area relatively frequently for around 2minutes. We are using the LVGL library from your example. I'll take a look at issue #30 you advise there

**Comment by Llgok** [comment]
@fundix 
You can try using the [GFX\_AXS15231B\_Image](https://github.com/Xinyuan-LilyGO/T-Display-S3-Long/tree/master/examples/GFX_AXS15231B_Image) program to see if it mistakenly triggers without any finger touch.

**Comment by malexnurazhar** [comment]
Im having the same issue. Random touch in the same area without touching.

**Comment by fundix** [comment]
@malexnurazhar 
> Im having the same issue. Random touch in the same area without touching.

Do you use WiFi in your project? Try disabling it and test if the random clicking remains. In my project, it seems that when I removed the code that manages the wifi, the clicking either disappeared or hasn't appeared yet. But with wifi it was quite partial.

**Comment by malexnurazhar** [comment]
No wifi no bluetooth. Just the display and touch. i tried using interrupts as suggested to only read i2c when theres touch. Didnt really affect anything. I suspect its something to do with the power. Its very sensitive. The issue got worse with a noisy power supply. Its better when powered by battery. But problem still persist every few seconds.

**Comment by malexnurazhar** [comment]
> @malexnurazhar
> 
> > Im having the same issue. Random touch in the same area without touching.
> 
> Do you use WiFi in your project? Try disabling it and test if the random clicking remains. In my project, it seems that when I removed the code that manages the wifi, the clicking either disappeared or hasn't appeared yet. But with wifi it was quite partial.
> 
> update: Update: No, so the problem will be as someone else wrote with what is displayed on the screen. The all white area causes me to click even without wifi.
> 
> [107915][I][Display.cpp:297] my_touchpad_read(): [Display] fingers_number: 2, touch event 8 [108065][I][Display.cpp:297] my_touchpad_read(): [Display] fingers_number: 2, touch event 8 [108215][I][Display.cpp:297] my_touchpad_read(): [Display] fingers_number: 2, touch event 8 [108365][I][Display.cpp:297] my_touchpad_read(): [Display] fingers_number: 2, touch event 8 ...

I suspect something to do with the power supply. Its a very power sensitive display. Not sure if my theory makes sense. When its white, a small voltage is applied to the pixel to turn all the film to transparent. This small voltage might have leak to the capacitive panel causing it to detect touch input.

**Comment by malexnurazhar** [comment]
But mine is clicking randomly even with all black.

**Comment by Llgok** [comment]
> But mine is clicking randomly even with all black.

@malexnurazhar 
Could you provide us with your complete project testing code? This would be more helpful in assisting us to identify any issues.

**Comment by fundix** [comment]
I'll try to get the project to compile without our other libraries

display ok
<img width="571" alt="Screenshot 2024-08-15 at 12 12 33" src="https://github.com/user-attachments/assets/f50cccb1-0410-412a-9666-e6f7a8f3cf84">

display not ok
<img width="580" alt="Screenshot 2024-08-15 at 13 00 01" src="https://github.com/user-attachments/assets/351f5266-ffb0-41fb-bfed-b71d266558a7">

[196146][I][Display.cpp:297] my_touchpad_read(): [Display] fingers_number: 2, touch event 8
[196249][I][Display.cpp:297] my_touchpad_read(): [Display] fingers_number: 2, touch event 8
[196353][I][Display.cpp:297] my_touchpad_read(): [Display] fingers_number: 2, touch event 8
[196505][I][Display.cpp:297] my_touchpad_read(): [Display] fingers_number: 2, touch event 8

**Comment by fundix** [comment]
> > But mine is clicking randomly even with all black.
> 
> @malexnurazhar Could you provide us with your complete project testing code? This would be more helpful in assisting us to identify any issues.

@Llgok I added you to our repository branch Touch-test
https://github.com/Liftor-GmbH/Liftor-Ortex-Remote/tree/Touch-test

the display goes off after about 30sec sometimes it lights up again

[872553][I][Display.cpp:190] display_wakeup_event_handler(): [Display] wakeup_event_handler - screen pressed
[872570][I][Display.cpp:296] my_touchpad_read(): [Display] fingers_number: 1, touch event 4
[906839][I][Display.cpp:296] my_touchpad_read(): [Display] fingers_number: 1, touch event 8
[906852][I][Display.cpp:296] my_touchpad_read(): [Display] fingers_number: 0, touch event 0
[906861][I][Display.cpp:190] display_wakeup_event_handler(): [Display] wakeup_event_handler - screen pressed

**Comment by Llgok** [comment]
@fundix 
I tested the code you provided and recorded a video for you to take a look at. It seems that there was no accidental screen wake-up trigger during the 30-second screen-off test.

https://github.com/user-attachments/assets/809dd550-169e-4253-8fd2-a9b791a67b71

**Comment by Llgok** [comment]
I will move the test output information function down to the triggering range.
![1](https://github.com/user-attachments/assets/501bbc97-87c6-405b-afb6-0bb3550e38f5)

**Comment by malexnurazhar** [comment]
All 3 of my unit is showing the same issue even with filters. So i disabled the touch input at the problematic area for the time being.
If(pointX > 500 && pointY > 100)return;
Will wait for the new ones to come. To see if its a faulty unit.

**Comment by fundix** [comment]
> @fundix I tested the code you provided and recorded a video for you to take a look at. It seems that there was no accidental screen wake-up trigger during the 30-second screen-off test.
> 
>  touch.test.mp4

@Llgok 

Thank you for the test. The problem for me arises later The display turns itself off after 30sec (the time can be changed in the swipe up settings) and lights up at different time intervals.  I shortened the video, it took around 1min till the light up.

<img width="877" alt="Screenshot 2024-08-19 at 12 10 39" src="https://github.com/user-attachments/assets/f56e534e-e6f7-44da-be2d-81d567859aef">

https://github.com/user-attachments/assets/8b63a1b2-9621-4260-8207-32054bb687ef

**Comment by malexnurazhar** [comment]
> All 3 of my unit is showing the same issue even with filters. So i disabled the touch input at the problematic area for the time being. If(pointX > 500 && pointY > 100)return; Will wait for the new ones to come. To see if its a faulty unit.

I got the new unit and the random touch isnt there on the new unit. Does it mean the one with the random touch has hardware issue??. Now the new unit has a new issue, i need to manually press boot and reset every single time to upload my code.

**Comment by Llgok** [comment]
@fundix 
You can try adding the following lines for power initialization parameters in your code:

Touch-test branch
liftor-remote.cpp[39,0]
```C++
bool WriteC8D8(uint8_t c, uint8_t d)
{
    if (Wire.write(c) == false)
    {
        log_e("->Write(c) fail");
        return false;
    }
    if (Wire.write(d) == false)
    {
        log_e("->Write(d) fail");
        return false;
    }
    return true;
}

bool IIC_WriteC8D8(uint8_t device_address, uint8_t c, uint8_t d)
{
    Wire.beginTransmission(device_address);
    if (WriteC8D8(c, d) == false)
    {
        log_e("->WriteC8D8(c, d) fail");
        return false;
    }
    if (!(Wire.endTransmission()) == false)
    {
        log_e("->EndTransmission() fail");
        return false;
    }
    return true;
}
```

Touch-test branch
liftor-remote.cpp[87,0]
```C++
    //Disable the ILIM pin and set the input current limit to maximum.
    IIC_WriteC8D8(0x6A, 0x00, 0B00111111);
    //Turn off the BATFET without using the battery.
    IIC_WriteC8D8(0x6A, 0x09, 0B01100100); 
```
![2](https://github.com/user-attachments/assets/6fcf199c-8de0-452f-b454-c5be418ba2ef)
![1](https://github.com/user-attachments/assets/1af88c56-672e-4a54-9c81-bad58d879f05)

Check if the random touch issue disappears after adding the above code.

**Comment by fundix** [comment]
@Llgok 
Looks like that solves the problem. Thank you very much.

**Comment by Eddy-the-Git** [comment]
I had the same issue with random touches when using esp_now_send(..). The power initialization parameters mentioned here fixed it for me. Thank you!
